types/module/module.go

```diff
@@ -320,15 +320,37 @@ func (m *Manager) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, genesisData
 }
 
 // ExportGenesis performs export genesis functionality for modules
-func (m *Manager) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) map[string]json.RawMessage {
+func (m *Manager) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec, modulesToExport []string) map[string]json.RawMessage {
 	genesisData := make(map[string]json.RawMessage)
-	for _, moduleName := range m.OrderExportGenesis {
-		genesisData[moduleName] = m.Modules[moduleName].ExportGenesis(ctx, cdc)
+	if len(modulesToExport) == 0 {
+		for _, moduleName := range m.OrderExportGenesis {
+			genesisData[moduleName] = m.Modules[moduleName].ExportGenesis(ctx, cdc)
+		}
+	} else {
+		// verify modules exists in app, so that we don't panic in the middle of an export
+		if err := m.checkModulesExists(modulesToExport); err != nil {
+			panic(err)
+		}
+
+		for _, moduleName := range modulesToExport {
+			genesisData[moduleName] = m.Modules[moduleName].ExportGenesis(ctx, cdc)
+		}
 	}
 
 	return genesisData
 }
 
+// checkModulesExists verifies that all modules in the list exist in the app
+func (m *Manager) checkModulesExists(moduleName []string) error {
```

Comment:
nit: we could just write a generics function instead. e.g. `SliceContains[T]`.

Reply:
I think it may be a tiny bit too generic? We will need a generic map comparable map too.

Reply:
ahh yeah, we'd have two type constraints, `T,K`, but maybe this is overkill then. Feel free to leave as-is 👍